sh/tests/grades_test.sh

```diff
@@ -5,65 +5,50 @@ IFS='
 script_dirS=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)
 
 challenge() {
+    args=$1
+    input=$2
 
-    declare -a equals=("Student1 90" "Student2 70" "Student3 50")
-    declare -a differents=("Student1 92" "Student2 75" "Student3 55" "Student4 25")
-
-    for e in "${equals[@]}"; do
-        name=$(echo "$e" | awk '{print $1}')
-        grade=$(echo "$e" | awk '{print $2}')
-        submitted=$(echo "$name $grade" | bash -c ""$script_dirS"/student/grades.sh 1")
-        expected=$(echo "$name $grade" | bash -c ""$script_dirS"/solutions/grades.sh 1")
-        diff <(echo "$submitted") <(echo "$expected")
-    done
-    for d in "${differents[@]}"; do
-        name=$(echo "$d" | awk '{print $1}')
-        grade=$(echo "$d" | awk '{print $2}')
-        submitted=$(echo "$name $grade" | bash -c ""$script_dirS"/student/grades.sh 1")
-        expected=$(echo "$name $grade" | bash -c ""$script_dirS"/solutions/grades.sh 1")
-        diff <(echo "$submitted") <(echo "$expected")
-    done
-
-    # Testing multiple students
     submitted=$(
-        bash -c ""$script_dirS"/student/grades.sh 3" <<EOF
-Bob
-90
-Alice
-75
-Eve
-55
+        bash -c ""$script_dirS"/student/grades.sh $args" <<EOF
+$input
 EOF
     )
     expected=$(
-        bash -c ""$script_dirS"/solutions/grades.sh 3" <<EOF
-Bob
-90
-Alice
-75
-Eve
-55
+        bash -c ""$script_dirS"/solutions/grades.sh $args" <<EOF
+$input
 EOF
     )
     diff <(echo "$submitted") <(echo "$expected")
```

Comment:
The tester returns a 0 even when the tests fail, it should return `1` or anything different than `0` instead.
You could add this just after the `diff`:
```bash
    if [ $? != 0 ]; then
        exit 1;
    fi
```
It will check if `diff` exited with a non 0 value, if so it means the two entries are different and the tester will `exit 1`.